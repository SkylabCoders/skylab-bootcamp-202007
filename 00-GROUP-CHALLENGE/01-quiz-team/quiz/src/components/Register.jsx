import React from "react";

function Register(props){
    return (
        <p>REGISTER PAGE</p>
    );
}

export default Register;