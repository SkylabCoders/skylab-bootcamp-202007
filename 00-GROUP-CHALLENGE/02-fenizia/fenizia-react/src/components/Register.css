.register-container {
  display: flex;
  flex-direction: column;
  height: 80vh;
  justify-content: center;
}
.register-container__wrap {
  margin: auto;
  width: 40vw;
  height: 40vh;
  text-align: center;
  border: 2px solid #333;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
}
.register-container__wrap__btn {
  height: 3vh;
  width: 10vw;
  border: none;
}
.register-container__wrap__btn:hover {
  background-color: #fff333;
}
