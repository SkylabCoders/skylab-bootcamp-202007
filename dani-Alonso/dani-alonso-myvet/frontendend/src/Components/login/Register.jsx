import React , {useState}from 'react';
import {
    StyleSheet,
    View,
    TextInput,
    TouchableOpacity,
    Text
} from "react-native"
import {createUser} from "../../actions/authActions"

export default function Register({navigation}) {
  const [username, setUsername] = useState('')
  const [firstName, setFirstname] = useState('')
  const [lastName, setLastname] = useState('')
  const [password, setPassword] = useState('')

  const regUser=event =>{
    event.preventDefault()
    const newUserParams = {username, firstName, lastName, password}
    createUser(newUserParams)
    
  }
    
      return (<>
        <View style={styles.container}>
        <View style={styles.inputView} >
          <TextInput  
            style={styles.inputText}
            placeholder="Username" 
            defaultValue= ""
            placeholderTextColor="#003f5c"
            onChangeText={ (username) => setUsername(username)}/>
        </View>
        <View style={styles.inputView} >
          <TextInput  
            style={styles.inputText}
            placeholder="First Name" 
            defaultValue= ""
            placeholderTextColor="#003f5c"
            onChangeText={ (firstName) => setFirstname(firstName)}/>
        </View>
        <View style={styles.inputView} >
          <TextInput  
            style={styles.inputText}
            placeholder="Last Name" 
            defaultValue= ""
            placeholderTextColor="#003f5c"
            onChangeText={ (lastName) => setLastname(lastName)}/>
        </View>
        <View style={styles.inputView} >
          <TextInput  
            style={styles.inputText}
            placeholder="Password" 
            defaultValue= ""
            secureTextEntry={true}
            placeholderTextColor="#003f5c"
            onChangeText={ (password) => setPassword(password)}/>
        </View>
       
        <TouchableOpacity style={styles.loginBtn}
        onPress={regUser}
        >
          <Text style={styles.loginText}
          >
          Register</Text>
        </TouchableOpacity>

        <TouchableOpacity>
          <Text style={styles.loginText}
          onPress={()=> navigation.navigate("Login")}>Login</Text>
        </TouchableOpacity>
      </View>
      </>);
}
  
  
  const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: 'white',
        alignItems: 'center',
        justifyContent: 'center',
      },
      inputView:{
        width:"80%",
        backgroundColor:"#E9E4F0",
        borderRadius:25,
        height:50,
        marginBottom:20,
        justifyContent:"center",
        padding:20,
      },
      inputText:{
          height:50,
          color:"#00AF80"
      },
      loginBtn:{
        width:"80%",
        backgroundColor:"#00AF80",
        borderRadius:25,
        height:50,
        alignItems:"center",
        justifyContent:"center",
        marginTop:40,
        marginBottom:10,
      }
  });