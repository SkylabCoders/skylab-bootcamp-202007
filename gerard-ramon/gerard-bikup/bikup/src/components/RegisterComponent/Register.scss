$light-bg-color: white;
$light-highlight-color: #eeeeee;
$main-color: #0c9;
$dark-highlight-color: #393e46;
$strava-button-bg-color: #fc4c02;
$strava-button-color: whitesmoke;
$standard-button-color: black;
$header-bg-color: #222831;

$light-register-bg: rgba(238, 238, 238, 0.6);

.register {
	width: 100%;
	height: 100%;
}

.register__logo {
	height: 40px;
	margin-right: 12px;
	cursor: pointer;
}

.register__upper {
	display: flex;
	flex-direction: row;
	align-items: center;

	padding: 12px;
}

.register__bikup {
	font-size: 20px;
	font-weight: bold;
	text-decoration: none;
	color: black;
}

.flex-spacer {
	display: flex;
	flex: 1;
}

.register__container {
	display: flex;
	flex-direction: column;
	align-items: center;
	margin-top: 24px;
}

.register__container::after {
	content: '';
	background-image: url('../../assets/landing1.jpeg');
	background-repeat: no-repeat;
	background-size: cover;
	background-position: center left;
	opacity: 0.6;
	top: 0;
	left: 0;
	bottom: 0;
	right: 0;
	position: absolute;
	z-index: -1;
}

.register__form {
	display: flex;
	flex-direction: column;
	align-items: center;
	width: 80%;
	padding: 12px;
	background-color: $light-register-bg;
	border-radius: 5px;
	margin-top: 12px;
}

.input__label {
	padding: 5px;
}

.register__input {
	height: 35px;
	width: 100%;
	border: 1px solid $light-highlight-color;
	border-radius: 5px;
	padding: 6px 24px;
	margin-bottom: 5px;
}

.input__error {
	outline: none;
	border: 1px solid red;
}

input:focus {
	outline: none;
	border: 1px solid $main-color;
}

.register__form__bottom {
	display: flex;
	flex-direction: row;
	align-items: center;
	justify-content: center;
	width: 100%;
	margin-top: 6px;
}

.register__external {
	display: flex;
	flex-direction: column;
	align-items: center;
	width: 80%;
	margin-top: 12px;
}

.login__warning {
	display: flex;
	flex-direction: row;
	justify-content: flex-end;
	width: 100%;
	padding: 0 24px;
	font-weight: 500;
	height: 20px;
	color: red;
	font-size: 14px;
}

.warning__error {
	color: red;
}

.warning__warning {
	color: yellow;
}

.form-register__button {
	padding: 12px;
	border: none;
	background-color: $main-color;
	border-radius: 5px;
	font-weight: bold;
	color: whitesmoke;
	font-size: 18px;
	width: 100%;
	margin-top: 6px;
	cursor: pointer;
	transition: all 0.2s ease-in-out;
}

.form-register__button:hover {
	transform: scale(1.03);
	color: black;
}

.register__btn {
	height: 45px;
	margin: 6px 0;
}

.loading {
	display: flex;
	flex-direction: row;
	align-items: center;
	justify-content: center;
}

.loading__img {
	height: 30px;
}

.no-loading {
	display: flex;
	flex-direction: row;
	align-items: center;
	justify-content: center;
}

.hidden {
	display: none;
}

.register__button--header {
	padding: 12px;
	border: none;
	background-color: $main-color;
	border-radius: 5px;
	font-weight: bold;
	color: whitesmoke;
	font-size: 18px;
	width: 100px;
	margin-right: 24px;
	cursor: pointer;
	transition: all 0.2s ease-in-out;
}

.register__button--header:hover {
	transform: scale(1.03);
	color: black;
}

@media only screen and (min-width: 900px) {
	.mobile {
		display: none;
	}

	.register {
		width: 400px;
		margin: auto;
	}
}
