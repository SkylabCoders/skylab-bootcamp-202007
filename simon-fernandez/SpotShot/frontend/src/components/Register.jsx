import React, { useEffect, useState } from "react";
import {
  TextInput,
  View,
  Text,
  TouchableOpacity,
  KeyboardAvoidingView,
} from "react-native";
import stylesRegister from "../styles/register-style";
import { signUpUser } from "../actions/authActions";
import authStore from "../stores/authStore";

export default function Register() {
  const [username, setUsername] = useState("");
  const [firstName, setFirstName] = useState("");
  const [lastName, setLastName] = useState("");
  const [password, setPassword] = useState("");
  const [confirmPassword, setConfirmPassword] = useState("");
  const [message, setMessage] = useState(null);

  function checkPasswordmatch() {
    if (password !== confirmPassword) {
      setMessage("password does not match");
    } else {
      signUpUser(username, password, firstName, lastName);
    }
  }
  function onChange() {
    setMessage(authStore.getErr());
  }
  useEffect(() => {
    authStore.addChangeListener(onChange);
    return () => authStore.removeChangeListener(onChange);
  }, []);

  return (
    <KeyboardAvoidingView
      behavior="padding"
      style={stylesRegister.inputContainer}
    >
      <Text style={stylesRegister.headerText}>Register </Text>
      <View>
        {message && <Text style={stylesRegister.errorMessage}>{message}</Text>}
        <Text style={stylesRegister.inputTextHeader}>Username</Text>
        <TextInput
          multiline={false}
          autoFocus
          style={stylesRegister.inputText}
          onChangeText={(text) => setUsername(text)}
          value={username}
        />

        <Text style={stylesRegister.inputTextHeader}>First Name </Text>
        <TextInput
          multiline={false}
          autoFocus
          style={stylesRegister.inputText}
          onChangeText={(text) => setFirstName(text)}
          value={firstName}
        />

        <Text style={stylesRegister.inputTextHeader}>Last Name </Text>
        <TextInput
          multiline={false}
          autoFocus
          style={stylesRegister.inputText}
          onChangeText={(text) => setLastName(text)}
          value={lastName}
        />

        <Text style={stylesRegister.inputTextHeader}>Password</Text>
        <TextInput
          multiline={false}
          secureTextEntry
          style={stylesRegister.inputText}
          onChangeText={(text) => setPassword(text)}
          value={password}
        />
        <Text style={stylesRegister.inputTextHeader}> Confirm password</Text>
        <TextInput
          multiline={false}
          secureTextEntry
          style={stylesRegister.inputText}
          onChangeText={(text) => setConfirmPassword(text)}
          value={confirmPassword}
        />
        <TouchableOpacity
          style={stylesRegister.submitButtonContainer}
          onPress={() => checkPasswordmatch(password, confirmPassword)}
        >
          <Text style={stylesRegister.submitButton}>Confirm</Text>
        </TouchableOpacity>
      </View>
    </KeyboardAvoidingView>
  );
}
