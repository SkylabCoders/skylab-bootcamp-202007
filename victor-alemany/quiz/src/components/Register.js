import React from "react";

function Register(props){
    return;
}

export default Register;