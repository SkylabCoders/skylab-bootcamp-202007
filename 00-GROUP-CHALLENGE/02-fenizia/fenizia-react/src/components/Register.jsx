import React, { useState } from "react";
import { signIn } from "../actions/authActions";
import TextInput from "./TextInput";
import "./Register.css";
import authStore from "../stores/authStore";

function Register() {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [signInError, setSignInError] = useState(authStore.getSignError());
  const [path, setPath] = useState("");
  function onFieldChange(value, setValue) {
    setValue(""+value);

  }
  function signInchecker() {
  
    if (password) {
      signIn(email, password);
      setPath('/login')
    }else{
      alert('The password hst to be 6 characters or more')
    }
    console.log(signInError);
    // fetch(signIn(email, password))
    //   .then((response) => {
    //     console.log("sign in worked");
    //     console.log(response);
    //   })
    //   .catch((error) => console.log(error));
  }
  return (
    <>
      <main className="register-container">
        <div className="register-container__wrap">
          <span>Register</span>
          <div>
            <TextInput
              type="text"
              name="username"
              placeholder="mail"
              value={email}
              onChange={(event) => onFieldChange(event.target.value, setEmail)}
            />
          </div>
          <div>
            <TextInput
              type="password"
              name="password"
              placeholder="Password"
              value={password}
              onChange={(event) =>
                onFieldChange(event.target.value, setPassword)
              }
            />
          </div>
          <div>
            <a href={path}>
            <button
              className="register-container__wrap__btn"
              onClick={() => signInchecker()}
            >
              Sign in
            </button>
              
            </a>
          </div>
        </div>
      </main>
    </>
  );
}
export default Register;
