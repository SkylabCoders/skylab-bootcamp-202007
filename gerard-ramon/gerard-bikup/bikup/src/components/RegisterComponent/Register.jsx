import React from 'react';
import { NavLink, useHistory } from 'react-router-dom';

import authStore from '../../stores/authStore';

import { createUserWithMail } from '../../actions/authActions';

import './Register.scss';

function Register() {
	const history = useHistory();

	function handleLogoClick() {
		history.push('/');
	}

	function getFormElements() {
		const emailInputElement = document.getElementsByClassName(
			'register__email'
		)[0];
		const passwordInputElement = document.getElementsByClassName(
			'register__password'
		)[0];
		const repeatInputElement = document.getElementsByClassName(
			'register__repeat-password'
		)[0];
		const submitButtonElement = document.getElementsByClassName(
			'form-register__button'
		)[0];
		const loadingElement = document.getElementsByClassName('loading')[0];
		const noLoadingElement = document.getElementsByClassName(
			'no-loading'
		)[0];
		const warningMessageElement = document.getElementsByClassName(
			'login__warning'
		)[0];

		const formElements = {
			emailInputElement,
			passwordInputElement,
			repeatInputElement,
			submitButtonElement,
			loadingElement,
			noLoadingElement,
			warningMessageElement,
		};

		return formElements;
	}

	function disableForm(formElements) {
		formElements.emailInputElement.disabled = true;
		formElements.passwordInputElement.disabled = true;
		formElements.repeatInputElement.disabled = true;
		formElements.submitButtonElement.disabled = true;

		formElements.loadingElement.style.display = 'flex';
		formElements.noLoadingElement.style.display = 'none';
	}

	function enableForm(formElements) {
		formElements.emailInputElement.disabled = false;
		formElements.passwordInputElement.disabled = false;
		formElements.repeatInputElement.disabled = false;
		formElements.submitButtonElement.disabled = false;

		formElements.loadingElement.style.display = 'none';
		formElements.noLoadingElement.style.display = 'flex';
	}

	async function handleSubmit() {
		const formElements = getFormElements();

		const email = formElements.emailInputElement.value;
		const password = formElements.passwordInputElement.value;
		const repeat = formElements.repeatInputElement.value;

		if (email && password && repeat) {
			if (password !== repeat) {
				formElements.warningMessageElement.innerHTML =
					'Passwords must match';
			} else {
				disableForm(formElements);

				const formData = {
					email,
					password,
				};
				await createUserWithMail(formData);
				const userResponse = authStore.getAuthUser();
				if (typeof userResponse === 'string') {
					formElements.warningMessageElement.innerHTML = userResponse;
					enableForm(formElements);
				} else {
					history.replace('/bikes');
				}
			}
		}
	}

	function checkRepeatPassword() {
		const password = document.getElementsByClassName(
			'register__password'
		)[0].value;
		const repeat = document.getElementsByClassName(
			'register__repeat-password'
		)[0].value;

		if (password === repeat) {
			document.getElementsByClassName('login__warning')[0].innerHTML = '';
		} else {
			document.getElementsByClassName('login__warning')[0].innerHTML =
				'Passwords must match';
		}
	}

	return (
		<>
			<div className='landing__header desktop'>
				<NavLink to='/' className='header__linkhome'>
					<img
						className='logo'
						src='https://cdn.discordapp.com/attachments/692420285143711814/693437226146594876/LogoGerili.png'
						alt='logo'
					/>
					<p className='appName'>bikUP</p>
				</NavLink>

				<div className='flex-spacer'></div>
				<NavLink to='/login' className='landing-header-login'>
					<button className='register__button--header'>Log In</button>
				</NavLink>
			</div>
			<div className='register__upper mobile'>
				<img
					src='https://cdn.discordapp.com/attachments/692420285143711814/693437226146594876/LogoGerili.png'
					alt='bikup logo'
					className='register__logo'
					onClick={handleLogoClick}
				/>
				<NavLink to='/' className='register__bikup'>
					bikUP
				</NavLink>
				<div className='flex-spacer'></div>
				<NavLink className='login__button' to='/login'>
					Log In
				</NavLink>
			</div>
			<div className='register'>
				<div className='register__container'>
					<h1>Create an account</h1>

					<form
						name='registerForm'
						action='/api/auth/mail'
						className='register__form'
						method='POST'
					>
						<label
							htmlFor='email'
							className='input__label'
							required
						>
							Email
						</label>
						<input
							type='email'
							className='register__email register__input'
							name='email'
							placeholder='Enter your email...'
							required
						/>
						<label htmlFor='password' className='input__label'>
							Password
						</label>
						<input
							type='password'
							className='register__password register__input'
							name='password'
							placeholder='Enter your password...'
							pattern='^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#$%^&*_=+-]).{8,12}$'
							required
						/>
						<label
							htmlFor='repeat-password'
							className='input__label'
						>
							Repeat Password
						</label>
						<input
							type='password'
							className='register__repeat-password register__input'
							name='repeat_password'
							placeholder='Repeat the password...'
							pattern='^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#$%^&*_=+-]).{8,12}$'
							required
							onChange={(event) => {
								event.preventDefault();
								checkRepeatPassword();
							}}
						/>
						<div className='login__warning '></div>
						<div className='register__form__bottom'>
							<button
								className='form-register__button'
								type='submit'
								onClick={(event) => {
									event.preventDefault();
									handleSubmit();
								}}
							>
								<div className='loading hidden'>
									<img
										src='https://trello-attachments.s3.amazonaws.com/5f4cb639a6f5eb1005114de4/5f5753c458a8b552f891bb81/af512a8cb3c1285000d1191fdaaa670c/Spinner-1s-200px_(1).gif'
										alt='loading...'
										className='loading__img'
									/>
									<p>Loading...</p>
								</div>
								<div className='no-loading' Create Account>
									Create an account
								</div>
							</button>
						</div>
					</form>

					<div className='register__separator'>
						<div className='line'></div>
						<span className='bold'>Or</span>
						<div className='line'></div>
					</div>

					<div className='register__external'>
						<button className='register__btn strava'>
							<div className='btn__logo'>
								<img
									src='https://trello-attachments.s3.amazonaws.com/5f4cb639a6f5eb1005114de4/5f53a18bf970328231db4f61/2f6af477333d585dfbfc19a14da9f857/strava-2.svg'
									alt='google logo'
								/>
							</div>
							<div className='btn__text'>
								register with Strava
							</div>
						</button>

						<button className='register__btn google'>
							<div className='btn__logo'>
								<img
									src='https://trello-attachments.s3.amazonaws.com/5f4cb639a6f5eb1005114de4/5f53a18bf970328231db4f61/99c6b9ab576aa9e324980db26e5fc7c0/google-logo.png'
									alt='google logo'
								/>
							</div>
							<div className='btn__text'>
								register with Google
							</div>
						</button>
					</div>
				</div>
			</div>
		</>
	);
}

export default Register;
