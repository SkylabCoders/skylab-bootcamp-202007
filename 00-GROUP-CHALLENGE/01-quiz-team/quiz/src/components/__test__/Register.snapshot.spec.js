import React from 'react';
import renderer from 'react-test-renderer';
import Register from './../Register';

const tree = renderer.create(<Register />).toJSON();

describe('Register snapshot', () => {
    test('should match', () => {
        expect(tree).toMatchSnapshot();
    })
})