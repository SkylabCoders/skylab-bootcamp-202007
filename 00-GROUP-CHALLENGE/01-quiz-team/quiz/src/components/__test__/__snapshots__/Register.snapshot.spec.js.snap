// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register snapshot should match 1`] = `
<p>
  REGISTER PAGE
</p>
`;
